src/index.js

```diff
@@ -187,24 +187,57 @@ class OurGroceriesCard extends LitElement {
   renderBody() {
     const body = (this.entity.attributes.shopping_lists || []).map(list => {
       let addingItem = (this.showAddItems[list.id] || {});
-      
+
       let isOpen = this.openedLists[list.id];
       let listDetails = isOpen && this.listItems[list.id];
       if (this.config.expanded) this.openList(list);
 
-      return html`
-        <tr>
-          <td class='td td-name pointer'>
-            <ha-icon icon="mdi:plus" @click="${event => this.toggleNewItem(event, list.id)}"></ha-icon>
-            <span @click=${() => this.openList(list)}>${list.name}</span>
-          </td>
-          <td class='td td-count'>
-            ${list.activeCount} 
-          </td>
-        </tr>
-        ${addingItem.show ? this.renderNewItem(addingItem, list): null}
-        ${isOpen && listDetails ? this.renderList(listDetails) : null}
-      `;
+      let listsHaveBeenSpecified = (this.config.show_lists.length > 0);

```

Comment:
pull things like this out of the loop. they wont change each loop. make `const` if not being changed

Reply:
I've moved 'listsHaveBeenSpecified' & 'allowingEmptyLists' out of the loop.